### ISSUE TITLE
Collector-Mail-Url cannot mark E-Mails as sent, if the download is very slow

### ISSUE THREAD

**Issue by dmth** [certtools/intelmq#852]
When a download is very slow the E-Mail within the "Collector-Mail-Url" can not be marked as read.
In addition, the data is retrieved a second (third,...) time when the bot failed.

The collector needs to refresh the IMAP-Connection instead of runnning into a timeout.

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/intelmq/lib/bot.py", line 122, in start
    self.process()
  File
"/usr/lib/python3/dist-packages/intelmq/bots/collectors/mail/collector_mail_url.py",
line 93, in process
    mailbox.mark_seen(uid)
  File "/usr/lib/python3/dist-packages/imbox/__init__.py", line 54, in mark_seen
    self.connection.uid('STORE', uid, '+FLAGS', '(\\Seen)')
  File "/usr/lib/python3.4/imaplib.py", line 816, in uid
    typ, dat = self._simple_command(name, command, *args)
  File "/usr/lib/python3.4/imaplib.py", line 1133, in _simple_command
    return self._command_complete(name, self._command(name, *args))
  File "/usr/lib/python3.4/imaplib.py", line 958, in _command_complete
    raise self.abort('command: %s => %s' % (name, val))
imaplib.abort: command: UID => Disconnected for inactivity.
2017-01-02 17:50:25,545 - Shadowserver-Open-MDNS - INFO - Current
Message(event): None.
2017-01-02 17:50:25,546 - Shadowserver-Open-MDNS - INFO - Bot will restart
in 15 seconds.
2017-01-02 17:50:40,558 - Shadowserver-Open-MDNS - INFO - Bot woke up
2017-01-02 17:50:40,559 - Shadowserver-Open-MDNS - INFO - Trying to start
processing again.
2017-01-02 17:50:40,630 - Shadowserver-Open-MDNS - INFO - Reading email report
2017-01-02 17:50:40,630 - Shadowserver-Open-MDNS - INFO - Downloading report
from https://dl.shadowserver.org/REDACTED
2017-01-02 17:52:41,955 - Shadowserver-Open-MDNS - INFO - Report downloaded.
```

**Comment by dmth** [comment]
BTW: This behaviour does **most likely** not happen once #854 is solved.

Nevertheless it **will** happen if the downloaded dataset is huge and does take a long time.

So we've got at least two options here:

1) Reconnect to the server if the client was disconnected and, if necessary, mark the E-Mail selected above as seen. This requires to identify the correct E-Mail. Maybe a UUID of the mail can be stored for a second selection.

2) Keep the connection open, somehow. The IDLE command might work for this, but I'm not sure if it is the right way to do it.

I vote for 1) as disconnects can always happen, due to other reasons. It is the more robust choice.

**Comment by wagner-certat** [comment]
1) yes, I think this is a good solution

2) that would require that a thread does send the IDLEs, right?

**Comment by wagner-certat** [comment]
Are you working on this or plan to do so?

**Comment by wagner-certat** [comment]
IDLE is not a feasible option as imaplib and thus imbox do not support it, see martinrusev/imbox#1

**Comment by dmth** [comment]
> Are you working on this or plan to do so?

No, we are currently not.

**Comment by wagner-certat** [comment]
fix is in #1100
